packages/beacon-node/src/chain/stateCache/stateContextCache.ts

```diff
@@ -40,6 +45,10 @@ export class StateContextCache {
       return null;
     }
 
+    if (this.head?.stateRoot === rootHex) {
```

Comment:
when head state is not in `this.cache` it will not hit `this.head` check

this should be chained to line 43 above, this is why issue https://github.com/ChainSafe/lodestar/issues/4700 still happened